scvi/inference/posterior.py

```diff
@@ -224,62 +224,103 @@ def sample_scale_from_batch(self, n_samples, batchid=None, selection=None):
         bayes1 = get_bayes_factors(px_scale, all_labels, cell_idx=0, M_permutation=M_permutation,
                                    permutation=False, sample_pairs=sample_pairs)
         if all_stats is True:
-            bayes1_permuted = get_bayes_factors(px_scale, all_labels, cell_idx=0, M_permutation=M_permutation,
+            bayes1_permuted = get_bayes_factors(px_scale, all_labels, cell_idx=0,
+                                                M_permutation=M_permutation,
                                                 permutation=True, sample_pairs=sample_pairs)
-            bayes2 = get_bayes_factors(px_scale, all_labels, cell_idx=1, M_permutation=M_permutation,
+            bayes2 = get_bayes_factors(px_scale, all_labels, cell_idx=1,
+                                       M_permutation=M_permutation,
                                        permutation=False, sample_pairs=sample_pairs)
-            bayes2_permuted = get_bayes_factors(px_scale, all_labels, cell_idx=1, M_permutation=M_permutation,
+            bayes2_permuted = get_bayes_factors(px_scale, all_labels, cell_idx=1,
+                                                M_permutation=M_permutation,
                                                 permutation=True, sample_pairs=sample_pairs)
             mean1, mean2, nonz1, nonz2, norm_mean1, norm_mean2 = \
                 self.gene_dataset.raw_counts_properties(idx1, idx2)
             res = pd.DataFrame([bayes1, bayes1_permuted, bayes2, bayes2_permuted,
                                 mean1, mean2, nonz1, nonz2, norm_mean1, norm_mean2,
                                 px_scale_mean1, px_scale_mean2],
                                index=['bayes1', 'bayes1_permuted', 'bayes2', 'bayes2_permuted',
-                                      'mean1', 'mean2', 'nonz1', 'nonz2', 'norm_mean1', 'norm_mean2',
+                                      'mean1', 'mean2', 'nonz1', 'nonz2', 'norm_mean1',
+                                      'norm_mean2',
                                       'scale1', 'scale2'],
                                columns=self.gene_dataset.gene_names).T
             res = res.sort_values(by=['bayes1'], ascending=False)
             return res
         else:
-            return(bayes1)
+            return pd.Series(data=bayes1, index=self.gene_dataset.gene_names)
 
     @torch.no_grad()
     def one_vs_all_degenes(self, subset=None, cell_labels=None, min_cells=10,
-                           n_samples=None, M_permutation=None, output_file=False,
+                           n_samples=1000, M_permutation=None, output_file=False,
                            save_dir='./', filename='one2all'):
+        """
+
+        :param subset: MASK: subset of cells you are interested in
+        :param cell_labels: Labels of cells
+        :param min_cells: Ceil number of cells used to compute Bayes Factors
+        :param n_samples: see `differential_expression_score`
```

Comment:
You might have to copy over ? Is it good practice to see "go to xxx" ?

Reply:
Better to copy